### PR TITLE
[#466] Mapping order (project item) data to JIRA issue

### PR DESCRIPTION
**Pull request by michal-szostak**
# What is in this PR?

* Add custom fields mapping for JIRA
* Add custom field checks and suggestions to checked (and
  `rake jira:check` task)

Fixes: #466

### CODE REVIEW

File: app/models/jira/checker.rb [hunk 1]
```diff
@@ -141,6 +142,19 @@ def check_workflow_transitions!(issue)
...
   end
 
+  def check_custom_fields!
+    fields = self.client.Field.all
+
+    statuses = self.client.custom_fields.to_a.map do |field_name, field_id|
+      [field_name, !fields.find { |f| f.id == field_id && f.name.to_sym == field_name }.nil?]
+    end .to_h
+
+    unless statuses.find { |k, v| !v  }.nil?
```
Comment (mkasztelnik) [inline app/models/jira/checker.rb]: you can use `all?` or `any?` to return boolean value.

File: app/models/jira/client.rb [hunk 2]
```diff
@@ -30,14 +33,96 @@ def initialize
...
+    issue = self.Issue.build
+
+    fields = { summary: "Service order, #{project_item.project.user.first_name} " +
+                       "#{project_item.project.user.last_name}, " +
+                       "#{project_item.service.title}",
+              project: { key: @jira_project_key },
+              issuetype: { id: @jira_issue_type_id },
+    }
+
+    @custom_fields.reject { |k, v| !v || v.empty? }.each do |field_name, field_id|
```
Comment (mkasztelnik) [inline app/models/jira/client.rb]: You can use `blank?` method:

```ruby
nil.blank? # true
"".blank?  # true
[].blank? # true
```

File: app/models/jira/client.rb [hunk 3]
```diff
@@ -30,14 +33,96 @@ def initialize
...
+    when "CI-Surname"
+      project_item.project.user.last_name
+    when "CI-Email"
+      project_item.affiliation ? project_item.affiliation.email : ""
+    when "CI-DisplayName"
+      "#{project_item.project.user.first_name} #{project_item.project.user.last_name}"
+    when "CI-EOSC-UniqueID"
+      project_item.project.user.uid
+    when "CI-Institution"
+      project_item.affiliation ? project_item.affiliation.organization : ""
```
Comment (mkasztelnik) [inline app/models/jira/client.rb]: You can simplify it a bit:

```ruby
project_item.affiliation&.organization || ""
```

File: lib/jira/console_checker.rb [hunk 4]
```diff
@@ -63,12 +63,33 @@ def show_available_issue_states
       end
     end
 
+    def show_suggested_fields_mapping(mismatched_fields)
+      fields = @checker.client.Field.all.select { |f| f.custom }
+
+      puts "SUGGESTED MAPPINGS"
+      mismatched_fields.each do |field_name|
+        suggested_field_description = if (f = fields.find { |f| f.name == field_name.to_s })
```
Comment (mkasztelnik) [inline lib/jira/console_checker.rb]: shadowing outer local variable - `f`

File: app/models/jira/client.rb (hunk 3 above)
Comment (michal-szostak) [reply app/models/jira/client.rb]: that's handy

**Pull request merged**